File: CPP/Recursion/palindrome.cpp
#include <bits/stdc++.h>
#include <algorithm>
using namespace std;

// using recursion

bool checkPalindrome(int i, string &str){
    if(i>=str.length()/2){
        return true;
    }
    if(str[i] != str[str.length()-i-1])
        return false;

    return checkPalindrome(i+1, str);
}

int main(){
    string str = "MADAM";
    cout << checkPalindrome(0, str) << endl;
    //reverse(str.begin(), str.end());
    //cout << str;
}

File: python/practice/recursion/palindrome.py
def checkPalindrome(i, name):
    if i >= len(name):
        return True
    
    if name[i] != name[len(name) - i - 1]:
        return False

    return checkPalindrome(i+1, name)

name = "MADAM"
print(checkPalindrome(0, name))